certified-connectors/DocuSignDemo/script.csx

```diff
@@ -3007,6 +3083,340 @@ public class Script : ScriptBase
+      {
+        foreach (var customField in body["customFields"]["listCustomFields"] as JArray)
+        {
+          var required = "";
+          if (string.Equals(customField["required"].ToString(), "true"))
+          {
+            required = " *";
+          }
+          itemProperties["List Custom Fields:::" + customField["name"].ToString()] = new JObject
+          {
+            ["type"] = "string",
+            ["x-ms-summary"] = "List Custom Fields: " + required + customField["name"].ToString(),
+            ["enum"] = customField["listItems"]
+          };
+        }
+      }
+      if (body["customFields"]["textCustomFields"] != null)
+      {
+        foreach (var customField in body["customFields"]["textCustomFields"] as JArray)
+        {
+          var required = "";
+          if (string.Equals(customField["required"].ToString(), "true"))
+          {
+            required = " *";
+          }
+          itemProperties["Text Custom Fields:::" + customField["name"].ToString()] = new JObject
+          {
+            ["type"] = "string",
+            ["x-ms-summary"] = "Text Custom Fields: " + required + customField["name"].ToString()
+          };
+        }
+      }
+    }
+  }
+
+  private void ParseRecipientFields(Dictionary<string, JObject> recipientMapping, string[] keyArray, string value, string roleName)
+  {
+    if (!recipientMapping.ContainsKey(roleName))
+    {
+      JObject newRecipientObj = new JObject();
+      newRecipientObj["roleName"] = roleName;
+      recipientMapping[roleName] = newRecipientObj;
+    }
+    JObject recipientObj = recipientMapping[roleName];
+
+    if (keyArray.Length > 2)
+    {
+      if (!recipientObj.ContainsKey("tabs"))
+      {
+        recipientObj["tabs"] = new JObject();
+      }
+      var tabType = keyArray[1];
+      var tabLabel = keyArray[2];
+      if (string.Equals(tabType, "checkboxTabs", StringComparison.OrdinalIgnoreCase))
+      {
+        var tabObj = new JObject
+        {
+          ["tabLabel"] = tabLabel,
+          ["name"] = keyArray[3],
+          ["selected"] = value
+        };
+        JObject recipientTabs = (JObject) recipientObj["tabs"];
+        if (!recipientTabs.ContainsKey(tabType))
+        {
+          recipientTabs[tabType] = new JArray();
+        }
+        JArray recipientObjArray = (JArray) recipientTabs[tabType];
+        recipientObjArray.Add(tabObj);
+      }
+      else
+      {
+        var tabObj = new JObject
+        {
+          ["tabLabel"] = tabLabel,
+          ["value"] = value
+        };
+        JObject recipientTabs = (JObject) recipientObj["tabs"];
+        if (!recipientTabs.ContainsKey(tabType))
+        {
+          recipientTabs[tabType] = new JArray();
+        }
+        JArray recipientObjArray = (JArray) recipientTabs[tabType];
+        recipientObjArray.Add(tabObj);
+      }
+    }
+    if (string.Equals(keyArray[1], "Name", StringComparison.OrdinalIgnoreCase))
```

Comment:
Does this work?
switch (keyArray[1].ToLower())
{
    case "name":
        recipientObj["name"] = value;
        break;
    case "email":
        recipientObj["email"] = value;
        break;
    case "signing group":
        recipientObj["signingGroupId"] = value;
        break;
    case "in person signer":
        recipientObj["inPersonSignerName"] = value;
        break;
}

Reply:
I also added a `StringComparison.OrdinalIgnoreCase` also if we were to go with switch approach, we would need  if case for tabs parsing logic, as the if case for tabs logic is different that other fields.